### ISSUE TITLE
[Bug] GoogleMapsOverlay is missing the right click event

### ISSUE THREAD

**Issue by digitalkoicodes** [visgl/deck.gl#7661]
### Description

When using the Icon layer on google maps, right click events do not happen

### Flavors

- [ ] React
- [ ] Python/Jupyter notebook
- [ ] MapboxLayer
- [X] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

onClick to emit with right clicks having rightButton = true in the event

### Steps to Reproduce

Here's an example codepen: 

https://codepen.io/mjtworks/pen/abwNJVW 

The event should happen with right or left click

### Environment

- Framework version: 8.8.21

**Comment by Pessimistress** [comment]
@felixpalmer I think we may be able use [addDomListener](https://developers.google.com/maps/documentation/javascript/events#DomEvents) instead of map's own events

**Comment by felixpalmer** [comment]
There is a `rightclick` event, we just ignore it. The button pressed can be obtained via `event.srcEvent.domEvent.button`

**Comment by Pessimistress** [comment]
Fixed in v8.9.3.

**Comment by moonbe77** [comment]
I'm using  GoogleMapsOverlay with TS and i can't get around that ts error
```TS
import { GoogleMapsOverlay } from '@deck.gl/google-maps'
	overlay.setProps({
			// useDevicePixels: true,
			onClick: (layer, event) => {
				const isRightClick = event.srcEvent.domEvent.button === 2  // Property 'domEvent' does not exist on 
    type 'TouchEvent | MouseEvent | PointerEvent'.  Property 'domEvent' does not exist on type 'TouchEvent'.ts(2339)
}
```

any suggestion on how to get the correct type for the onClick > event, thanks
